test/unit/core/algorithm/configuration_test.cpp

```diff
@@ -222,58 +223,139 @@ TEST(configuration, remove_by_type_template)
     }
 }
 
-TEST(configuration, value_or_by_type)
+TEST(configuration, get_or_by_type)
 {
     seqan3::configuration cfg = bax{2.2} | bar{1};
 
     { // l-value
-        EXPECT_FLOAT_EQ(cfg.value_or<bax>(1.3), 2.2);
-        EXPECT_FLOAT_EQ(cfg.value_or<foo>(1.3), 1.3);
+        EXPECT_FLOAT_EQ(cfg.get_or(bax{1.3}).value, 2.2);
+        EXPECT_TRUE((std::same_as<decltype(cfg.get_or(bax{1.3})), bax &>));
+        EXPECT_EQ(cfg.get_or(foo{"test"}).value, "test");
     }
 
     { // const l-value
-        seqan3::configuration<bax, bar> const cfg_c{cfg};
-        EXPECT_FLOAT_EQ(cfg_c.value_or<bax>(1.3), 2.2);
-        EXPECT_FLOAT_EQ(cfg_c.value_or<foo>(1.3), 1.3);
+        EXPECT_FLOAT_EQ(std::as_const(cfg).get_or(bax{1.3}).value, 2.2);
+        EXPECT_TRUE((std::same_as<decltype(std::as_const(cfg).get_or(bax{1.3})), bax const &>));
+        EXPECT_EQ(std::as_const(cfg).get_or(foo{"test"}).value, "test");
```

Comment:
We checked before also the value of 1.3, why are we not doing that anymore? (Same question for the other tests.)

Reply:
Because the semantics changed. Before I could give any alternative, but now the type to look for is deduced by the type of the alternative we pass in. So in this case we use `foo<std::string>` and want to see if any `foo` specialisation is present. If not we return `foo<std::string>`.